### ISSUE TITLE
Support only 2 latest major go releases

### ISSUE THREAD

**Issue by davidhsingyuchen** [cweill/gotests#168]
## TL;DR

Raised a PoC [PR](https://github.com/davidhsingyuchen/gotests/pull/2), and it passed all the checks. If it looks good to you, I can raise a PR in this repository.

## Why

From [Go's release policy](https://go.dev/doc/devel/release#policy):

> Each major Go release is supported until there are two newer major releases.

As a result, unless under some special circumstances, I feel that it does not make much sense to support versions that are not even supported by official Go team.

## Context

As of now, [tests are failing](https://github.com/cweill/gotests/runs/5260950688?check_suite_focus=true) with the following error message:

> ../../../golang.org/x/tools/internal/typeparams/normalize.go:162:17: u.EmbeddedType undefined (type *types.Interface has no field or method EmbeddedType)

That is because [in Go1.10, `type *types.Interface has no field or method EmbeddedType`](https://github.com/golang/go/issues/31693).

I also noticed that we have `golang.org/x/tools v0.0.0-20191109212701-97ad0ed33101` in our [`go.mod`](https://github.com/cweill/gotests/blob/16a93f6eb6519118b1d282e2f233596a98dd7e96/go.mod#L3), and in that version, the whole `typeparams` directory does not even exist. From the line number in the error message above, the build job seems to use the [latest version](https://github.com/golang/tools/blob/897bd77cd7177f6d679eafa9767ff6ef9211458a/internal/typeparams/normalize.go#L162) instead of the one specified in `go.mod`, and why is that?

The Go version is 1.10 in the job, and [modules was introduced in 1.11](https://go.dev/doc/go1.11#modules), and I suspect that the `go get ./...` simply downloads the latest version because it just does not understand `go.mod`.

At this timepoint, I figured that maybe it makes more sense to support the latest releases (i.e., 1.17 and so on) instead of making effort to maintain releases that are too old.
